src/main/webapp/ts/profile.ts

```diff
@@ -0,0 +1,9 @@
+async function getProfile() {
+    const response = await fetch('/profile');
```

Comment:
consider separating the frontend changes for a separate pull request as this makes this change more focused and eaiser to review.

Reply:
Ok